File: Example/Example-Mac/ViewController.m
//
//  ViewController.m
//  Example-Mac
//
//  Created by Takeru Chuganji on 1/4/15.
//
//

#import "ViewController.h"
#import "HCImage+BPG.h"

@interface ViewController ()
@property (weak) IBOutlet NSImageView *imageView;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    NSString *filePath = [[NSBundle mainBundle] pathForResource:@"sample" ofType:@"bpg"];
    NSData *data = [NSData dataWithContentsOfFile:filePath];
    self.imageView.image = [NSImage imageWithBPGData:data];
}

- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];
    // Update the view, if already loaded.
}

@end

File: Example/Mac/Example/ViewController.swift
//
//  ViewController.swift
//  Example
//
//  Created by Takeru Chuganji on 11/11/15.
//  Copyright © 2015 Takeru Chuganji. All rights reserved.
//

import Cocoa

class ViewController: NSViewController {

    @IBOutlet private weak var imageView: NSImageView?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.imageView?.image = NSBundle.mainBundle().pathForResource("sample", ofType: "bpg")
            .flatMap { NSData(contentsOfFile: $0) }
            .flatMap { NSImage(BPGData: $0) }
    }
}
